lib/bolt/inventory/group.rb

```diff
@@ -18,12 +18,18 @@ class Group
       GROUP_KEYS = DATA_KEYS + %w[name groups targets]
       CONFIG_KEYS = Bolt::Config::INVENTORY_OPTIONS.keys
 
-      def initialize(input, plugins)
+      def initialize(input, plugins, all_group: false)
         @logger = Bolt::Logger.logger(self)
         @plugins = plugins
 
         input = @plugins.resolve_top_level_references(input) if @plugins.reference?(input)
 
+        if all_group && input.key?('name')
+          raise ValidationError.new("Top-level group 'all' cannot specify name '#{input['name']}'", nil)
```

Comment:
I don't understand why we want to prevent this - it's currently allowed at the top level, and will get overwritten by the merge in `inventory.rb`. This seems unnecessary and risky?